packages/swapper/src/swappers/JupiterSwapper/endpoints.ts

```diff
@@ -0,0 +1,144 @@
+      contractAddress && isManualReceiveAddress
+        ? await adapter.createAssociatedTokenAccountInstruction({
+            from,
+            to: tradeQuote.receiveAddress,
+            tokenId: contractAddress,
+          })
+        : { instruction: undefined, destinationTokenAccount: undefined }
+
+    const maybeSwapResponse = await getJupiterSwapInstructions({
+      apiUrl: jupiterUrl,
+      fromAddress: from,
+      toAddress: isManualReceiveAddress ? destinationTokenAccount?.toString() : undefined,
+      rawQuote: tradeQuote.rawQuote,
+      // Shared account is not supported for simple AMMs
+      useSharedAccounts:
+        tradeQuote.rawQuote.routePlan.length > 1 && isManualReceiveAddress ? true : false,
+    })
+
+    if (maybeSwapResponse.isErr()) {
+      const error = maybeSwapResponse.unwrapErr()
+      const cause = error.cause as AxiosError<any, any>
+      throw Error(cause.response!.data.detail)
+    }
+
+    const { data: swapResponse } = maybeSwapResponse.unwrap()
+
+    const convertJupiterInstruction = (instruction: Instruction): TransactionInstruction => ({
+      ...instruction,
+      keys: instruction.accounts.map(account => ({
+        ...account,
+        pubkey: new PublicKey(account.pubkey),
+      })),
+      data: Buffer.from(instruction.data, 'base64'),
+      programId: new PublicKey(instruction.programId),
+    })
+
+    const instructions: TransactionInstruction[] = [
+      ...swapResponse.setupInstructions.map(convertJupiterInstruction),
+      convertJupiterInstruction(swapResponse.swapInstruction),
+    ]
+
+    if (createTokenAccountInstruction) {
+      instructions.unshift(createTokenAccountInstruction)
+    }
+
+    if (swapResponse.cleanupInstruction) {
+      instructions.push(convertJupiterInstruction(swapResponse.cleanupInstruction))
+    }
+
+    const getFeeData = async () => {
+      const { chainNamespace } = fromAssetId(step.sellAsset.assetId)
+
+      switch (chainNamespace) {
+        case CHAIN_NAMESPACE.Solana: {
+          const sellAdapter = assertGetSolanaChainAdapter(step.sellAsset.chainId)
+          const getFeeDataInput: GetFeeDataInput<KnownChainIds.SolanaMainnet> = {
+            // Simulates a self-send
+            to: from,
+            value: '0',
+            chainSpecific: {
+              from,
+              addressLookupTableAccounts: swapResponse.addressLookupTableAddresses,
+              instructions,
+            },
+          }
+          return await sellAdapter.getFeeData(getFeeDataInput)
+        }
+
+        default:
+          throw new Error('Unsupported chainNamespace')
+      }
+    }
+
+    const { fast } = await getFeeData()
+
+    const solanaInstructions = instructions.map(instruction =>
+      adapter.convertInstruction(instruction),
+    )
+
+    const buildSwapTxInput: BuildSendApiTxInput<KnownChainIds.SolanaMainnet> = {
+      to: '',
+      from,
+      value: '0',
```

Comment:
We absolutely never send any values as a regular send, but we always use jupiter instructions, so we zero out every values as we were sending an SPL token